src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,170 @@
+
 package core.basesyntax;
 
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_FOR_EXCEPTION = "Index: ";
+    private static final String SIZE_FOR_EXCEPTION = ", Size: ";
+    private static final int NUMBER_TO_SUBTRACT = 1;
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> next;
+
+        Node(T value) {
+            this.value = value;
+        }
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        validateIndexForAdd(index);
+        Node<T> newNode = new Node<>(value);
+        if (index == 0) {
+            addFirst(newNode);
+        } else if (index == size) {
+            addLast(newNode);
+        } else {
+            Node<T> previousNode = getNode(index - NUMBER_TO_SUBTRACT);
+            newNode.next = previousNode.next;
+            previousNode.next = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
```

Comment:
Just in case, for me this variant is more readable
```suggestion
       for (T value : list) {
            add(value);
```